### PR TITLE
operations: add Munge page

### PR DESCRIPTION
**Pull request by heitorPB**
Describe how to create a new account to query the Munge key and add two
examples on how to do that.

Online at: https://omnivector-solutions.github.io/osd-documentation/testing/operations/munge.html

Docs for https://github.com/omnivector-solutions/slurm-charms/pull/144/

### CODE REVIEW

File: operations/munge.rst [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+================
+Munge operations
+================
+
+The ``slurmctld`` charm contains an internal `etcd3 <https://etcd.io>`_ server
+that, among other uses, stores the `Munge <https://github.com/dun/munge/>`_
+key.
+
+The etcd server has authentication enabled by default, to protect the Munge
+key. In order to be able to retrieve the Munge key, one needs an etcd account.
```
Comment (dusktreader) [inline operations/munge.rst:12]:
```suggestion
key. In order to be able to retrieve the Munge key, an etcd account is required.
```

File: operations/munge.rst [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+OSD provides Juju action ``etcd-create-munge-account`` on the ``slurmctld``
+charm to automate the creation of such account.
```
Comment (dusktreader) [inline operations/munge.rst:14]:
```suggestion
charm to automate the creation of such an account.
```

File: operations/munge.rst [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+
+Creating an etcd account
+========================
+
+To create an etcd account to query the Munge key requires a desired username
+and password:
```
Comment (dusktreader) [inline operations/munge.rst:21]:
```suggestion
To create an etcd account to query the Munge key requires a username and password to be specified:
```

File: operations/munge.rst [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+       enqueued: 2022-05-12 17:57:41 +0000 UTC
+       started: 2022-05-12 17:57:44 +0000 UTC
+
+
+Querying the munge key
+======================
+
+The etcd server is running on the ``slurmctld`` node and listening on the port
+``2379``. The Munge key is stored at ``munge/key`` entry. Please find below
+examples on how to query the secret key.
```
Comment (dusktreader) [inline operations/munge.rst:46]:
```suggestion
``2379``. The Munge key is stored at ``munge/key`` entry. The examples below
demonstrate how to query the secret key.
```

File: operations/munge.rst [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+
+Using Bash
+----------
+
+It is possible to get the Munge key using Bash, with ``curl`` and ``jq``. Firs,
```
Comment (dusktreader) [inline operations/munge.rst:51]:
```suggestion
It is possible to get the Munge key using Bash, with ``curl`` and ``jq``. First,
```

File: operations/munge.rst [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+<https://opendev.org/openstack/etcd3gw>`_ as a starting point:
+
+.. code-block:: python
+
+   import logging
+
+   from etcd3gw.client import Etcd3Client
+   from etcd3gw.exceptions import Etcd3Exception
+
+   logger = logging.getLogger(__name__)
```
Comment (dusktreader) [inline operations/munge.rst:93]: For example code, it might be best to exclude the logger.

**Review by dusktreader — CHANGES_REQUESTED** [review]
Nice docs! Added a few small comments for you.

File: operations/munge.rst (hunk 6 above)
Comment (heitorPB) [reply operations/munge.rst:93]: I tried to make the code as small as possible, did not think about removing the logger. Good idea.

File: operations/munge.rst [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+   user=theusername
+   pass=aVerySafePassword!
+   host=10.107.185.126
+   port=2379
+
+   token=$(curl -L -s -X POST "$host:$port/v3/auth/authenticate" -d '{"name":"'"$user"'", "password":"'"$pass"'"}' | jq .token)
+   token=${token//\"/}
+   key=$(printf munge/key | base64)
+
+   munge_key=$(curl -L -s -X POST "$host:$port/v3/kv/range" -H "Authorization: ${token}" -d '{"key":"'$key'"}' | jq .kvs[0].value)
```
Comment (jamesbeedy) [inline operations/munge.rst:66]:
```suggestion
   token=$(curl -L -s -X POST "$host:$port/v3/auth/authenticate" -d '{"name":"'"$user"'", "password":"'"$pass"'"}' | jq .token | tr -d '"')
   key=$(printf munge/key | base64)

   munge_key=$(curl -L -s -X POST "$host:$port/v3/kv/range" -H "Authorization: ${token}" -d '{"key":"'$key'"}' | jq .kvs[0].value | tr -d '"')
```

File: operations/munge.rst [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
...DRWbE1jdGErcFZuR2Q2Zk1vYmdBVDd1eHJSMUJuUDAzUHhhTXE4YzlzNjVmNzlDUXpaZkt1V3drYm1pTmxOL3lsaU5ueDNnMjJOQ2JhVGgvcUdWdVB3Yk94bU9yMkNGTFgvbjJuV1U2ZkhGK285QUpROVpVa1VqWkt4WWJLUUlxdG9Ga29KWnMxKzRGa0FGSUNSMEk4YkNZSTAyU3hIZkx4cGpJWTNuY1IvQUx2OUlHSkVjTjNWVHRnVlJ5UlRWeTZUYXEweFZxT2JGWHlBUGwybWltS3RVZWlSNEc5RTZtL0w1ZGdIUVA0TkRRYzNqVnRBeTRDSzFXQUZhVE0yNWVEai9zOUNuV3lBOVRPekJwbVYvZE9kN0xCR3JFNE9wM3NQSUg4Nk1TTkpNaFY3TFl2dmhJT2FYdUlJSTAwZjAweVVENUV4Y0hVOEdMT1VmUDlMTkFCS1FiMFhDWVVXS1NaTWtSUUx1UzI2MGxuWFhBUkIxd2xLWVo0dTdZZ2IzWnE3WGh2L3BqblZOV2FvUjR0dWJBQlJBS2JiRUxJMlZmazNpNTRtdkxvdnRyN1hXRzMwUGNIbXRQRE1hdFliWlQ5TFVDTUZ6NzZ3MjFlK3ZhY3cvREZLd0ZwNEdIV29BRzRSYUxORHBxM1FjVzZ0WXdVd0FpbXFDNitwblJhTjNsZ1ExNXpCNWo4MGQzTUJMaHZnZDVweExhS0ZoeHFRWmU2QXdGQ0xWTFJBY2Q0b0Zkb3RNMzNGUUVGYjFDV2tFNUNMVFQwZU1yYm1zWFdsMVlWdVlkWW0vYWc4bkpYdC9VMDJNeHV0bG80N1Z6aXhDSmdvUzd3dmFCaXhzL1Q4MlJzWlhSSWNvN2Y1cVdkdW0wMm5kT0FJK2NtbUZjd1Z4b1hnZ2hvYVIzbEJ6M3A1NUZmc0pteVVpci9GUHVTekR5eG1DS3A0eUs5aUxmakwxL3VPK2ltK0VQdUNSWHgwRS9aL1BSZkJuQ29jbk5RUlZlV0lSNWtDK1piK2FlN0ZzRVhEazVJcHU5eDRZcm5KUHhSTGY0L3JVTVBRRGEzaDB3TlU2N3BwL1JuN1VlWHRCNXdLanorUUJWNGI5aEljb0U0bHpHRVdUTkg2QXlTb1NNc0tYblQ1VDliWk9SRTFsVzdnZVY4VHptK0hTbjhDNWJmMzA4NmhFRE81dG1EcmlTdz09"
```
Comment (jamesbeedy) [inline operations/munge.rst:73]:
```suggestion
   NzI5dVpXcysrWlEvYjNzWUFNbGluaWxITm44eGlLRStLWWtpTzRNNHIwT3lKV05sandVRFVsTHZTTjRkQ2dkWXA1R0t6VGk4Nks3blZvaHBXVy9EMEE1TTNOdGlpMjF0SEN4QW1zTm1VdFYxa01tYTd3UlVEcGxGRnBGYmR6MkZwa3daUGtreW5YUzY1U000dEptdWdBSGV5eGg1eHM1Q2Ryc3l3VERQL1VTOXNDdm5TalBncGtPdTNUc2xMQmVVcEtzYnJ6cUtTbTZLaFBlZWNZYXEwTGMwTzRJYnNqTEpNbTZEWXpQbkU5UGFSL2ZMdDMxU29PRjZsTHgyMzRpaCtmZm5MODlSNXc3SjAvcnVxZGZvU3NlQkNvekRHMlRjZG1LeXRPWUgxNndqUWdPUW96eHJ2SVFmUllQRlBDVkxHMTlHUndDSzBWRDRWR09CRGRFZEtwUTl5Q2QxcjFYUzhjTkdhTUNuRHFQenJsTDBNbzlncldUU3g4ZTZHTnhOeDZWT2FvL3E3SmFXWllMd1lLRHpkUHZ5bkUzVXBpS2lBTlNQbDd2a1JVYTFYNDNKUWFQdS8yTGtybUczOWUzK2tpS2JVRjJWMjV3aExEeHA2a2d1NDRWbE1jdGErcFZuR2Q2Zk1vYmdBVDd1eHJSMUJuUDAzUHhhTXE4YzlzNjVmNzlDUXpaZkt1V3drYm1pTmxOL3lsaU5ueDNnMjJOQ2JhVGgvcUdWdVB3Yk94bU9yMkNGTFgvbjJuV1U2ZkhGK285QUpROVpVa1VqWkt4WWJLUUlxdG9Ga29KWnMxKzRGa0FGSUNSMEk4YkNZSTAyU3hIZkx4cGpJWTNuY1IvQUx2OUlHSkVjTjNWVHRnVlJ5UlRWeTZUYXEweFZxT2JGWHlBUGwybWltS3RVZWlSNEc5RTZtL0w1ZGdIUVA0TkRRYzNqVnRBeTRDSzFXQUZhVE0yNWVEai9zOUNuV3lBOVRPekJwbVYvZE9kN0xCR3JFNE9wM3NQSUg4Nk1TTkpNaFY3TFl2dmhJT2FYdUlJSTAwZjAweVVENUV4Y0hVOEdMT1VmUDlMTkFCS1FiMFhDWVVXS1NaTWtSUUx1UzI2MGxuWFhBUkIxd2xLWVo0dTdZZ2IzWnE3WGh2L3BqblZOV2FvUjR0dWJBQlJBS2JiRUxJMlZmazNpNTRtdkxvdnRyN1hXRzMwUGNIbXRQRE1hdFliWlQ5TFVDTUZ6NzZ3MjFlK3ZhY3cvREZLd0ZwNEdIV29BRzRSYUxORHBxM1FjVzZ0WXdVd0FpbXFDNitwblJhTjNsZ1ExNXpCNWo4MGQzTUJMaHZnZDVweExhS0ZoeHFRWmU2QXdGQ0xWTFJBY2Q0b0Zkb3RNMzNGUUVGYjFDV2tFNUNMVFQwZU1yYm1zWFdsMVlWdVlkWW0vYWc4bkpYdC9VMDJNeHV0bG80N1Z6aXhDSmdvUzd3dmFCaXhzL1Q4MlJzWlhSSWNvN2Y1cVdkdW0wMm5kT0FJK2NtbUZjd1Z4b1hnZ2hvYVIzbEJ6M3A1NUZmc0pteVVpci9GUHVTekR5eG1DS3A0eUs5aUxmakwxL3VPK2ltK0VQdUNSWHgwRS9aL1BSZkJuQ29jbk5RUlZlV0lSNWtDK1piK2FlN0ZzRVhEazVJcHU5eDRZcm5KUHhSTGY0L3JVTVBRRGEzaDB3TlU2N3BwL1JuN1VlWHRCNXdLanorUUJWNGI5aEljb0U0bHpHRVdUTkg2QXlTb1NNc0tYblQ1VDliWk9SRTFsVzdnZVY4VHptK0hTbjhDNWJmMzA4NmhFRE81dG1EcmlTdz09
```

**Pull request merged**